label_studio/ml/models.py

```diff
@@ -277,100 +331,13 @@ def predict_tasks(self, tasks):
-        ml_api_result = self.api.make_predictions(tasks_ser, self.project)
-        if ml_api_result.is_error:
-            logger.info(f'Prediction not created for project {self}: {ml_api_result.error_message}')
-            return
-
-        if not (isinstance(ml_api_result.response, dict) and 'results' in ml_api_result.response):
-            logger.info(f'ML backend returns an incorrect response, it should be a dict: {ml_api_result.response}')
-            return
-
-        responses = ml_api_result.response['results']
-
-        if len(responses) == 0:
-            logger.warning(f'ML backend returned empty prediction for project {self}')
-            return
-
-        # ML Backend doesn't support batch of tasks, do it one by one
-        elif len(responses) == 1 and len(tasks) != 1:
-            logger.warning(
-                f"'ML backend '{self.title}' doesn't support batch processing of tasks, "
-                f'switched to one-by-one task retrieval'
-            )
-            instances = [self.predict_one_task(task, model_version=model_version) for task in tasks]
-            return instances
-
-        # wrong result number
-        elif len(responses) != len(tasks_ser):
-            logger.warning(f'ML backend returned response number {len(responses)} != task number {len(tasks_ser)}')
-
-        predictions = []
-        for task, response in zip(tasks_ser, responses):
-            if 'result' not in response:
-                logger.info(
-                    f"ML backend returns an incorrect prediction, it should be a dict with the 'result' field:"
-                    f' {response}'
-                )
-                return
-            predictions.append(
-                {
-                    'task': task['id'],
-                    'result': response['result'],
-                    'score': response.get('score'),
-                    'model_version': response.get('model_version', self.model_version),
-                }
-            )
+        predictions = self._get_predictions_from_ml_backend(tasks_ser)
         with conditional_atomic(predicate=db_is_not_sqlite):
             prediction_ser = PredictionSerializer(data=predictions, many=True)
             prediction_ser.is_valid(raise_exception=True)
             instances = prediction_ser.save()
         return instances
 
-    def predict_one_task(self, task, model_version):
-        if not model_version:
-            self.update_state()
-            if self.not_ready:
-                logger.debug(f'ML backend {self} is not ready to predict {task}')
-                return
-
-        if task.predictions.filter(model_version=self.model_version).exists():
-            # prediction already exists
-            logger.info(
-                f'Skip creating prediction with ML backend {self} for task {task}: model version '
-                f'{self.model_version} is up-to-date'
-            )
-            return
```

Comment:
what about these 2 things? they are missed in the new code

Reply:
I don't think they are lost:

- 1st "if": https://github.com/HumanSignal/label-studio/blob/fb-RND-97/label_studio/ml/models.py#L316
- 2nd "if": https://github.com/HumanSignal/label-studio/blob/fb-RND-97/label_studio/ml/models.py#L327

am I missing something?